### ISSUE TITLE
Fix S3655 FP: Types with implicit conversion operator

### ISSUE THREAD

**Issue by hankovich** [SonarSource/sonar-dotnet#7625]
### Description

False Positive S3655 is reported for types with implicit casts.

### Repro steps

```csharp
using System;

public struct Result<T>
{
    public T Value { get; set; }

    public Result(T val)
    {
        Value = val;
    }

    public static implicit operator Result<T>(T value) => new Result<T>(value);
}

public class Sample
{
    public static Result<DateTimeOffset?> ParseTimestamp()
    {
        return (DateTimeOffset?)null; // Noncompliant - FP
    }
}

```

### Expected behavior

No warning is reported.

### Actual behavior

Warning is reported.

### Related information

* C#/VB.NET Plugins version: 9.5.0.73987
* Visual Studio version: lVisual Studio 2022 (17.6.5)
* MSBuild / dotnet version: .NET 7
* Operating System: Windows 10

**Comment by zsolt-kolbay-sonarsource** [comment]
Hi @hankovich. I tried to reproduce it with no success.

```cs
public class Result<T>
{
    public T Value { get; set; }

    public Result(T val)
    {
        Value = val;
    }

    public static implicit operator Result<T>(T value) => new Result<T>(value);
}

public class Sample
{
    public static Result<DateTimeOffset?> ParseTimestamp()
    {
        return (DateTimeOffset?)null; // <- S3655 is not reported here
    }
}
```

Could you provide a small self-containing code sample where the FP is raised?

**Comment by hankovich** [comment]
Hi, sure. Just change `public class Result<T>` to `public struct Result<T>`

**Comment by zsolt-kolbay-sonarsource** [comment]
Thank you. Confirmed as False Positive. I'll update your comment with the modified code sample.

**Comment by gregory-paidis-sonarsource** [comment]
This has been fixed by [this ticket](https://sonarsource.atlassian.net/browse/NET-857).
(Only sonar members can access the link, as the backlog is private)
You will see the change in the next release.
